### ISSUE TITLE
The loss value when the model converges

### ISSUE THREAD

**Issue by yangyyt** [jishengpeng/WavTokenizer#20]
When the model converges to a relatively good situation, how much loss will be trained? In the current training code, the generator loss and the discriminator loss are？ The generation effect of my first epoch is very poor. The generator loss is ~70 and the discriminator loss is ~ 10.  Is this normal?

**Comment by jishengpeng** [comment]
> When the model converges to a relatively good situation, how much loss will be trained?

When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.

A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.

**Comment by dyyoungg** [comment]
> > When the model converges to a relatively good situation, how much loss will be trained?
> 
> When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> 
> A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.

What's the difference between loss and generator total loss in the pbar as follows
```
Epoch 0:   0%|   | xx/xx[00:15<10,  7.53s/it, loss=75.8, v_num=1, generator/total_loss=128.0, discriminator/total=11.90]
```

I checked the training step code, but found no definition of loss variable.
```
# train generator
if optimizer_idx == 1:
   ..............
   loss = (
                loss_gen_mp
                + self.hparams.mrd_loss_coeff * loss_gen_mrd
                + loss_fm_mp
                + self.hparams.mrd_loss_coeff * loss_fm_mrd
                + self.mel_loss_coeff * mel_loss
                + 1000 * commit_loss
                + loss_dac_1
                + loss_dac_2
            )

    self.log("generator/total_loss", loss, prog_bar=True)
    self.log("mel_loss_coeff", self.mel_loss_coeff)
    self.log("generator/mel_loss", mel_loss)
    self.log("commit_loss", commit_loss)
```

**Comment by yangyyt** [comment]
> loss

Thank you for your reply. Currently, 36 epochs have been trained using libriTTS data, and the generator loss has reached ~40. It may take some more training. After testing the current model, all reconstructed audio spectrograms will have horizontal lines at low frequencies. I don’t know if the author has encountered this before.

**Comment by yangyyt** [comment]
> > When the model converges to a relatively good situation, how much loss will be trained?
> 
> When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> 
> A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.


Thank you for your reply. Currently, 36 epochs have been trained using libriTTS data, and the generator loss has reached ~40. It may take some more training. After testing the current model, all reconstructed audio spectrograms will have horizontal lines at low frequencies. I don’t know if the author has encountered this before.

**Comment by handsomelys** [comment]
> > When the model converges to a relatively good situation, how much loss will be trained?
> 
> When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> 
> A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.

As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?

**Comment by jishengpeng** [comment]
> Thank you for your reply. Currently, 36 epochs have been trained using libriTTS data, and the generator loss has reached ~40. It may take some more training. After testing the current model, all reconstructed audio spectrograms will have horizontal lines at low frequencies. I don’t know if the author has encountered this before.

The absence of the expected result may be attributed to an incorrect configuration of parameters, suggesting a need to revisit and adjust the settings to ensure optimal performance

**Comment by jishengpeng** [comment]
> > > When the model converges to a relatively good situation, how much loss will be trained?
> > 
> > 
> > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> 
> As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?

This is a normal phenomenon. To monitor the performance of the model, it is sufficient to focus on the validation set results in TensorBoard, as well as the overall loss during the training process.

**Comment by handsomelys** [comment]
> > > > When the model converges to a relatively good situation, how much loss will be trained?
> > > 
> > > 
> > > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> > 
> > 
> > As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?
> 
> This is a normal phenomenon. To monitor the performance of the model, it is sufficient to focus on the validation set results in TensorBoard, as well as the overall loss during the training process.

Does the overall loss here refer to generator/total_loss in Tensorboard? Because there is no steady downward trend in the total of discriminator (in fact, it is growing)

**Comment by jishengpeng** [comment]
> > > > > When the model converges to a relatively good situation, how much loss will be trained?
> > > > 
> > > > 
> > > > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > > > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> > > 
> > > 
> > > As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?
> > 
> > 
> > This is a normal phenomenon. To monitor the performance of the model, it is sufficient to focus on the validation set results in TensorBoard, as well as the overall loss during the training process.
> 
> Does the overall loss here refer to generator/total_loss in Tensorboard? Because there is no steady downward trend in the discriminator/total (in fact, it is growing)

Observing the loss of the generator component

**Comment by handsomelys** [comment]
> > > > > > When the model converges to a relatively good situation, how much loss will be trained?
> > > > > 
> > > > > 
> > > > > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > > > > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> > > > 
> > > > 
> > > > As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?
> > > 
> > > 
> > > This is a normal phenomenon. To monitor the performance of the model, it is sufficient to focus on the validation set results in TensorBoard, as well as the overall loss during the training process.
> > 
> > 
> > Does the overall loss here refer to generator/total_loss in Tensorboard? Because there is no steady downward trend in the discriminator/total (in fact, it is growing)
> 
> Observing the loss of the generator component

Thanks for reply!

**Comment by handsomelys** [comment]
> > > > > > When the model converges to a relatively good situation, how much loss will be trained?
> > > > > 
> > > > > 
> > > > > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > > > > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> > > > 
> > > > 
> > > > As the number of training steps increases, the generator loss decreases steadily, but the discriminator loss is improving, with relatively large jitter. For example, my discriminator loss in epoch 1 is 10, but it becomes 11 in epoch 2. Do you have any clues?
> > > 
> > > 
> > > This is a normal phenomenon. To monitor the performance of the model, it is sufficient to focus on the validation set results in TensorBoard, as well as the overall loss during the training process.
> > 
> > 
> > Does the overall loss here refer to generator/total_loss in Tensorboard? Because there is no steady downward trend in the discriminator/total (in fact, it is growing)
> 
> Observing the loss of the generator component

I also want a reference value for the converged generator loss (because the historical dialogue of the issue seems to only give the discriminator a converged loss reference value)

**Comment by VJJJJJJ1** [comment]
> Thank you for your reply. Currently, 36 epochs have been trained using libriTTS data, and the generator loss has reached ~40. It may take some more training. After testing the current model, all reconstructed audio spectrograms will have horizontal lines at low frequencies. I don’t know if the author has encountered this before.

How many steps does your model begin to converge? When I trained on the libiritts dataset, the first 20K step models still hadn't converged (total loss was about 120-130). Do we need to wait any longer? My configuration is as follows, trained on 2 V100:
```
data:
  class_path: decoder.dataset.VocosDataModule
  init_args:
    train_params:
      filelist_path: ./WavTokenizer-main/data/libritts/train.txt
      sampling_rate: 24000
      num_samples: 24000
      batch_size: 16  # 20
      num_workers: 8

    val_params:
      filelist_path: ./WavTokenizer-main/data/libritts/val.txt
      sampling_rate: 24000
      num_samples: 24000
      batch_size: 5   # 10
      num_workers: 8
```
thank you for your reply!

**Comment by VJJJJJJ1** [comment]
> > > When the model converges to a relatively good situation, how much loss will be trained?
> > 
> > 
> > When the model's **final loss** approximates 25, the audio reconstruction quality is satisfactory.
> > A discriminator loss of 10 is acceptable, whereas a generator loss exceeding 70 is relatively high. The current loss status also depends on the data size per epoch. An epoch consisting of 10,000 hours is unusual, whereas 500 hours per epoch is considered normal.
> 
> Thank you for your reply. Currently, 36 epochs have been trained using libriTTS data, and the generator loss has reached ~40. It may take some more training. After testing the current model, all reconstructed audio spectrograms will have horizontal lines at low frequencies. I don’t know if the author has encountered this before.

same quetion. have you solved it?
